src/xenia/gpu/dxbc_shader_translator.cc

```diff
@@ -435,12 +435,19 @@ void DxbcShaderTranslator::StartVertexOrDomainShader() {
                                                   : dxbc::Dest::R(0, 0b0111),
                  dxbc::Src::VDomain(0b000110));
         if (register_count() >= 2) {
-          // Copy the primitive index to r1.x as a float.
+          // Copy the primitive index + offset to r1.x as a float.
           uint32_t primitive_id_temp =
               uses_register_dynamic_addressing ? PushSystemTemp() : 1;
           in_primitive_id_used_ = true;
+          a_.OpMov(dxbc::Dest::R(primitive_id_temp, 0b0001),
+                   dxbc::Src::VPrim());
+          a_.OpIAdd(dxbc::Dest::R(primitive_id_temp, 0b0001),
```

Comment:
Since this index appears to be treated like the vertex index, I think vPrim + offset should also be clamped to the min and the max indices from the registers, just like in vertex index loading. The clamping also restricts the index to the range with integers exactly representable as float32 (24 bits).